lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -412,109 +436,111 @@ withNetworkLayer tr network np project k = do
-                                    ( unsafeMkSublist $
-                                        (\(n, dc) -> ((txIndex, n), dc))
-                                            <$> zip [0 ..] dcerts
-                                    )
-                        ws <- BF.getAccountWithdrawals address
-                        blockEventsWithdraw <-
-                            forConcurrently ws \BF.AccountWithdrawal{..} -> do
-                                (bftx@BF.Transaction{_transactionIndex}, tx) <-
-                                    fetchTransaction network _accountWithdrawalTxHash
-                                txIndex <- _transactionIndex <?#> "_transactionIndex"
+                getAddressTxs bhFrom bhTo addrOrAcc = do
+                    traceWith tr $ MsgDiag "getAddressTxs"
+                    addrOrAcc & runBF . fmap fromBlockEvents . \case
+                        Left address -> do
+                            txs <- BF.allPages \paged ->
+                                empty404 $ BF.getAddressTransactions'
+                                    (BF.Address (encodeAddress @nd address))
+                                    paged
+                                    BF.Ascending
+                                    (Just $ headerToIndex bhFrom)
+                                    (Just $ headerToIndex bhTo)
+                            forConcurrently txs \BF.AddressTransaction{..} -> do
+                                (bftx, tx) <-
+                                    fetchTransaction tr network _addressTransactionTxHash
+                                txIndex <-
+                                    _addressTransactionTxIndex
+                                        <?#> "_addressTransactionTxIndex"
                                 txBlockEvents
                                     bftx
                                     (unsafeMkSublist [((txIndex, 0), tx)])
                                     (wholeList [])
-                        pure $ blockEventsRegDeleg <> blockEventsWithdraw
+                        Right account -> do
+                            let address = BF.Address $ encodeStakeAddress @nd account
+                            regTxHashes <-
+                                fmap BF._accountRegistrationTxHash
+                                    <$> empty404 (BF.getAccountRegistrations address)
+                            delTxHashes <-
+                                fmap BF._accountDelegationTxHash
+                                    <$> empty404 (BF.getAccountDelegations address)
+                            blockEventsRegDeleg <-
+                                forConcurrently (regTxHashes <> delTxHashes) \hash -> do
+                                    (tx@BF.Transaction{_transactionIndex}, dcerts) <-
+                                        concurrently
+                                            do BF.getTx hash
+                                            do fetchDelegation tr network hash
```

Comment:
```suggestion
                                            BF.getTx hash
                                            (fetchDelegation tr network hash)
```
(? My argument would be: `IO` actions are first-class values. All first-class values wear beautiful gowns made from pristine parentheses. 🧐 )

Reply:
Ok